Rerelease-chapter-1 (#226)
huggingface/smol-course: units/en/unit0/1.mdx

# Welcome to the 🤗 smol-course

![Fine-Tuning Course thumbnail](https://github.com/huggingface/smol-course/raw/main/banner.png)

Welcome to the comprehensive (and smollest) guide to **Fine-Tuning Language Models**!

This free course will take you on a journey, **from beginner to expert**, in understanding, implementing, and optimizing fine-tuning techniques for large language models.

This first unit will help you onboard:

* Discover the **course's syllabus**.
* **Get more information about the certification process and the schedule**.
* Get to know the team behind the course.
* Create your **account**.
* **Sign-up to our Discord server**, and meet your classmates and us.

Let's get started!

<Tip>

> This course is smol but fast! It's for devs looking to fast track their fine-tuning skills. If that's not you, check out the [LLM Course](https://huggingface.co/learn/llm-course/).

Reviewer: ```suggestion
This course is smol but fast! It's for software developers and engineers looking to fast track their LLM fine-tuning skills. If that's not you, check out the [LLM Course](https://huggingface.co/learn/llm-course/).
```

</Tip>

## What to expect from this course?

In this course, you will:

> * 📖 Study **instruction tuning, supervised fine-tuning, and preference alignment** in theory and practice.

Reviewer: stylistic opinion: the ** are unnecessary

(feel free to ignore)

* 🧑‍💻 Learn to **use established fine-tuning frameworks and tools** like TRL and Transformers.
* 💾 **Share your projects** and explore fine-tuning applications created by the community.
* 🏆 Participate in challenges where you will **evaluate your fine-tuned models against other students'.**
* 🎓 **Earn a certificate of completion** by completing assignments.

At the end of this course, you'll understand **how to fine-tune language models effectively and build specialized AI applications using the latest fine-tuning techniques**.

> Don't forget to [**sign up to the course!**](https://huggingface.co/mcp-course)

Reviewer: ```suggestion
Don't forget to [**sign up to the course!**](https://huggingface.co/llm-course)
```

assuming you meant this

* _Collaborations_: We're collaborating with Hugging Face's partners to give you the latest fine-tuning implementations and tools.
 
This **course is a living project, evolving with your feedback and contributions!** Feel free to open issues and PRs in GitHub, and engage in discussions in our Discord server.

## What's the syllabus?

Here is the **general syllabus for the course**. A more detailed list of topics will be released with each unit.

| Chapter | Topic                                       | Description                                                                                                            |
| ------- | ------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------- |
| 0       | Onboarding                                  | Set you up with the tools and platforms that you will use.                                                             |
| 1       | Instruction Tuning Fundamentals | Explain core concepts of instruction tuning, chat templates, and supervised fine-tuning. Show practical examples using TRL.       |

> | 2       | Preference Alignment              | Learn about RLHF, DPO, and other preference alignment techniques to make models follow human preferences. |

Reviewer: ```suggestion
| 2       | Preference Alignment              | Learn about RLHF, DPO, and other preference alignment techniques to make models follow human preferences like ChatGPT. |
```

| 3       | Parameter-Efficient Fine-Tuning               | Explore LoRA, QLoRA, and other efficient fine-tuning methods that reduce computational requirements.                                        |
| 4       | Evaluation and Analysis                                  | Learn how to evaluate fine-tuned models and analyze their performance across different metrics.                                        |

## What are the prerequisites?

To be able to follow this course, you should have:

* Basic understanding of AI and LLM concepts
* Familiarity with Python programming and machine learning fundamentals
* Experience with PyTorch or similar deep learning frameworks
* Understanding of transformers architecture basics

If you don't have any of these, don't worry. Check out the [LLM Course](https://huggingface.co/learn/llm-course/) to get started.

<Tip>

The above courses are not prerequisites in themselves, so if you understand the concepts of LLMs and transformers, you can start the course now!

</Tip>

## What tools do I need?

You only need 2 things:

> * _A computer_ with an internet connection and preferably GPU access (Google Colab works great).

Reviewer: and maybe HF jobs?
